src/content_scripts/autocorrect.js

```diff
@@ -0,0 +1,351 @@
+"use strict";
+
+const fractions = Object.freeze(["¼", "½", "¾", "⅐", "⅑", "⅒", "⅓", "⅔", "⅕", "⅖", "⅗", "⅘", "⅙", "⅚", "⅛", "⅜", "⅝", "⅞"]);
+
+const fractionValues = Object.freeze([1.0 / 4.0, 1.0 / 2.0, 3.0 / 4.0, 1.0 / 7.0, 1.0 / 9.0, 1.0 / 10.0, 1.0 / 3.0, 2.0 / 3.0, 1.0 / 5.0, 2.0 / 5.0, 3.0 / 5.0, 4.0 / 5.0, 1.0 / 6.0, 5.0 / 6.0, 1.0 / 8.0, 3.0 / 8.0, 5.0 / 8.0, 7.0 / 8.0]);
+
+const constants = Object.freeze(["π", "e"]);
+
+const constantValues = Object.freeze([Math.PI, Math.E]);
+
+// communication type
+const AUTOCORRECT_CONTENT = "autocorrectContent";
+
+let insertedText; // Last insert text
+let deletedText; // Last deleted text
+let lastTarget; // Last target
+let lastCaretPosition; // Last caret position
+
+let autocomplete = true;
+let quotes = true;
+let fracts = true;
+
+let autocorrections = {};
+
+let longest = 0;
+
+// Regular expressions
+let symbolpatterns = null;
+// Do not autocorrect for these patterns
+let apatterns = null;
+
+let emojiShortcodes = {};
+
+/**
+ * Get caret position.
+ *
+ * @param {Object} target
+ * @returns {number}
+ */
+function getCaretPosition(target) {
+	if (target.isContentEditable) {
+		target.focus();
+		let _range = document.getSelection().getRangeAt(0);
+		let range = _range.cloneRange();
+		let temp = document.createTextNode("\0");
+		range.insertNode(temp);
+		const caretposition = target.innerText.indexOf("\0");
+		temp.parentNode.removeChild(temp);
+		return caretposition;
+	}
+	else
+		return target.selectionStart;
+}
+
+/**
+ * Insert at caret.
+ * Adapted from: https://www.everythingfrontend.com/posts/insert-text-into-textarea-at-cursor-position.html
+ *
+ * @param {Object} target
+ * @param {string} atext
+ * @returns {void}
+ */
+function insertCaret(target, atext) {
```

Comment:
Yes, I made that change locally in my Chrome PR, which is depended on my Thunderbird PR, which is depended on this PR. 😀 I can make that change here instead if you would like. I made the change for the Chrome PR, since your current code does not work in Chrome and had to be changed anyway...

Reply:
Oh oh, hmm, _if_ you can split that maybe do integrate it here. (Cherry-pick git commits if that works or so, if we want to wake the [git evil](https://ohshitgit.com/) :see_no_evil: )

Reply:
@tdulcet Well… okay, so if we merge thuis first, I still would like a `TODO` comment in here at least for unifying the code later, so we do not forget it.

So could you please add this?

Reply:
I make the change here, as you requested: https://github.com/tdulcet/awesome-emoji-picker/commit/453592aa7a2de5a2e4ad323f33e35120c7390192#diff-05e47b062340648d069b0d345c6fe7a3R106.